src/Controls/src/Core/Handlers/Items/Android/MauiCarouselRecyclerView.cs

```diff
@@ -117,6 +117,7 @@ protected override void OnAttachedToWindow()
 		protected override void OnDetachedFromWindow()
 		{
 			ClearLayoutListener();
+			UnsubscribeCollectionItemsSourceChanged(ItemsViewAdapter);
```

Comment:
I wonder if this should move to teardown as well as unsubscribing from scrolled 

Is teardown called when the user navigates away?